libsolidity/analysis/DeclarationTypeChecker.cpp

```diff
@@ -390,8 +390,16 @@ void DeclarationTypeChecker::endVisit(VariableDeclaration const& _variable)
 		type = TypeProvider::withLocation(ref, typeLoc, isPointer);
 	}
 
-	_variable.annotation().type = type;
+	if (_variable.isConstant() && !type->isValueType())
+	{
+		bool allowed = false;
+		if (auto arrayType = dynamic_cast<ArrayType const*>(type))
+			allowed = arrayType->isByteArray();
+		if (!allowed)
+			m_errorReporter.declarationError(9259_error, _variable.location(), "Constants of non-value type not yet implemented.");
```

Comment:
I think we should split out the part that just fixes the ICE, i.e. the move from ``TypeChecker`` to ``DeclarationTypeChecker``. I'd hope that'd actually be enough... but if not, I guess we can just make this here a fatal error and at least that should do it? The other changes are a bit delicate - maybe it's fine, but I've not worked through them yet.

Reply:
OK. The error has to be fatal though, otherwise `fuzzer.exe < const_struct_with_mapping.sol` causes ICE.

I think we should fix that ICE still (in another PR).